File: Haskell/ABC/ABC002/ABC002_A_v01.hs
-- [contest]  ABC 002
-- [task]     A
-- [URL]      https://atcoder.jp/contests/abc002/tasks/abc002_1
-- [compiler] Haskell (GHC 7.10.3)
-- [status]   https://atcoder.jp/contests/abc002/submissions/6445476 : AC

-- modules to import --

import qualified Data.ByteString.Char8
import qualified Data.Maybe
import           Prelude

-- functions for this task --

tuplify2 (x:y:_) = (x,y)
tuplify2 _       = Prelude.undefined

read_Integer      = Prelude.fst . Data.Maybe.fromJust . Data.ByteString.Char8.readInteger
read_TupleInteger = tuplify2 . Prelude.map read_Integer . Data.ByteString.Char8.words

get_Integer      = read_Integer      <$> Data.ByteString.Char8.getLine
get_TupleInteger = read_TupleInteger <$> Data.ByteString.Char8.getLine

task_A :: (Integer, Integer) -> Integer
task_A (x, y)
        | x > y             = x
        | Prelude.otherwise = y

-- the main process is as follows --

main :: IO ()
main = do

        -- STEP.01
        -- read out the given data
        given_data <- get_TupleInteger

        -- STEP.02
        -- calculate & output the answer of this task
        Prelude.print $ task_A $ given_data

-- EOF --

File: C(GCC5.4.1)/ABC/ABC002/ABC002_A_v01.c
// AtCoder Beginner Contest 002 A
// [URL]        https://atcoder.jp/contests/abc002/tasks/abc002_1
// [compiler]   C (GCC 5.4.1)
// [submission] https://atcoder.jp/contests/abc002/submissions/4599210 : AC

#include <stdio.h>

int main (void)
{
	/* variables for main process */
	int X, Y;

	// STEP.01
	// read out the target integers
	scanf ("%d %d", &X, &Y);

	// STEP.02
	// output the bigger integer
	if (X > Y) { printf("%d\n", X); }
	else       { printf("%d\n", Y); }

	// STEP.TRUE_END
	return 0;
}